Aztec/Classes/Converters/NSAttributedStringToNodes.swift

```diff
@@ -59,17 +79,96 @@ class NSAttributedStringToNodes: Converter {
 }
 
 
+// MARK: - Dimensionality
+//
+private extension NSAttributedStringToNodes {
+
+    /// Attempts to merge the Right array of Element Nodes (Paragraph Level) into the Left array of Nodes.
+    ///
+    func merge(left: [ElementNode], right: [ElementNode]) -> Bool {
+        guard let (target, source) = findLowestMergeableNodes(left: left, right: right) else {
+            return false
+        }
+
+        target.children += source.children
+
+        return true
+    }
+
+
+    /// Returns the "Right Most" Blocklevel Node from a collection fo nodes.
+    ///
+    func rightMostParagraphStyleElements(from nodes: [Node]) -> [ElementNode] {
+        return paragraphStyleElements(from: nodes) { children in
+            return children.last
+        }
+    }
+
+
+    /// Returns the "Left Most" Blocklevel Node from a collection fo nodes.
+    ///
+    func leftMostParagraphStyleElements(from nodes: [Node]) -> [ElementNode] {
```

Comment:
If you agree with what I wrote above, this should turn into `leftmostParagraphStyleElements`.